src/Api/FilterCollectionFactory.php

```diff
@@ -0,0 +1,53 @@
+<?php
+
+/*
+ * This file is part of the API Platform project.
+ *
+ * (c) Kévin Dunglas <dunglas@gmail.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+declare(strict_types=1);
+
+namespace ApiPlatform\Core\Api;
+
+use Psr\Container\ContainerInterface;
+
+/**
+ * Filter collection factory.
+ *
+ * @author Baptiste Meyer <baptiste.meyer@gmail.com>
+ *
+ * @internal
+ */
+class FilterCollectionFactory
+{
+    private $filtersIds;
+
+    public function __construct(string ...$filtersIds)
```

Comment:
I prefer to use a raw array and use `string[]` in the PHPDoc, PHPStan will be able to check types using the PHPDoc.